[ROCK Pi S] Mainline 5.5.y (dev) target

ashthespy opened the pull request:
With RK3308 support landing in mainline, been trying to get the Rock Pi S up and running with 5.5.y. 

Kernel source is over at [`ashthespy/linux-rockchip`](https://github.com/ashthespy/linux-rockchip) and is just basically picking a few commits from different trees with some trial and error in between. 

What works:
* Console
* SD Card
* Wired network
* WiFi


PS: Suggestions welcome for improvement!

lanefu commented:
You can ignore the test failure.. it built against the wrong board.. my apologies..

lanefu commented:
I also didn't mean to close this :)

ashthespy commented:
@lanefu It still seems to be building against the wrong board?
```
[[0;32m o.k. [0m] Using config file [[0;33m /mnt/armbian/jenkins_runner/workspace/armbian-test-pr/build/userpatches/config-jenkins-kernel.conf [0m]
[[0;32m o.k. [0m] Command line: setting BOARD to [[0;33m nanopiair [0m]
[[0;32m o.k. [0m] Command line: setting BRANCH to [[0;33m current [0m]

```

lanefu commented:
@ashthespy yeah still wrong board... the CI is in an alpha state.. just ignore.. i'll probably turn the job off or at least ate notifications.

ashthespy commented:
@lanefu yeah, I was wondering how the CI decides what board to build, especially when you add new board/kernels like in the PR.

lanefu commented:
@ashthespy the code is better at matching on boards than kernels at the moment.. the targets are dynamically generated.... anyway.. its at an alpha state

[scripts if you are curious](https://github.com/armbian/ci-testing-tools/blob/master/jenkins_ci.sh)

ashthespy commented:
>@ashthespy the code is better at matching on boards than kernels at the moment.. the targets are dynamically generated.... anyway.. its at an alpha state

Okay, will ignore it for now!

Question about the best way to integrate the WiFi drivers (rtl8723ds) into the build system. For now, I added drivers from [`lwfinger/rtl8723ds`](https://github.com/lwfinger/rtl8723ds) into [`compilation-prepare.sh`](https://github.com/armbian/build/blob/020ddfcfa1b5f0d0973dc0cbabb79caaca515a5b/lib/compilation-prepare.sh#L265) - or is there a different recommend way?

lanefu commented:
> Question about the best way to integrate the WiFi drivers (rtl8723ds) into the build system. For now, I added drivers from [`lwfinger/rtl8723ds`](https://github.com/lwfinger/rtl8723ds) into [`compilation-prepare.sh`](https://github.com/armbian/build/blob/020ddfcfa1b5f0d0973dc0cbabb79caaca515a5b/lib/compilation-prepare.sh#L265) - or is there a different recommend way?

that's a good place to start.    

@igorpecovnik can you confirm this approach is okay?

igorpecovnik commented:
> can you confirm this approach is okay?

Yes, the plan is to move all wireless/special drivers there.

igorpecovnik commented:
I think we shell merge rockpis at DEV branch to the rockchip64-dev ... you know how to do it? Or I shell do it? Like rk3399 family is also merged to rockchip for current and dev.

ashthespy commented:
>I think we shell merge rockpis at DEV branch to the rockchip64-dev ... you know how to do it? Or I shall do it? Like rk3399 family is also merged to rockchip for current and dev.

Do you mean you'd like to base it on the `dev` branch from  `rockchip64_common` ([`tag:5.4.0-rc1-1120-ayufan`](https://github.com/ayufan-rock64/linux-mainline-kernel))? Not sure that will work, as we will need at least 5.5.rc3 (i.e [`077f46d`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=077f46db63652fb16e60bcd0d623acc2ef282065)) for RK3308 support. 

I did try initially to prepare a set of patches from 5.5.y, but was having trouble getting consistent patches, so took the easier route of a separate branch based on [`kernel/git/mmind/linux-rockchip.git`](https://git.kernel.org/pub/scm/linux/kernel/git/mmind/linux-rockchip.git/log/) for now.

igorpecovnik commented:
I think we will ditch that branch. Probably we can do it right now and switch to mainline 5.5.y? Let's wait for @armbian/boards-rockchip comments.

piter75 commented:
> I think we will ditch that branch. Probably we can do it right now and switch to mainline 5.5.y?

I am in favour of switching "dev" source for rockchip64.
Mainline 5.5.y is a good start IMHO and should not bring too much burden in maintaining after the initial migration, which also should not be too painfull as we already have "current" pointing to 5.4.x.

ashthespy commented:
Alright, so as I understand the idea is to:
1. Switch `dev` branch of `rockchip64` over to mainline 5.5.y
2. Ditch the `rockpis` family (for `dev` at least), and use `rockchip64` family instead 

Then i shall work on making a patch list based on 5.5.y from my current branch - wish me luck! :-)

ashthespy commented:
Didn't manage to prepare the patches, I got sidetracked and ported across the audio codecs. :sweat_smile: 
The i2s drivers should also work in theory, but could only test the boards analogue outputs.  

Also merged in mainline 5.5, so I hope patchsets should be easier - I must confess thought that I have quite low experience with the entire patch workflow.. 
Either way, should I wait until  `rockchip64` switches over to mainline, and keep working on my branch for now?

piter75 commented:
> Either way, should I wait until `rockchip64` switches over to mainline

@ashthespy `rockchip64-dev` is now based on mainline's v5.5.y: https://github.com/armbian/build/pull/1781 :)

Regarding the patches... they could simply mirror your commits.
I would do a single one with the content's of Akash's introductory patch and then additional ones with each of them adding some atomic feature like sound, tsadc and so on.

ashthespy commented:
>@ashthespy rockchip64-dev is now based on mainline's v5.5.y: #1781 :)

Ah, I missed #1781 - that does indeed make it easier, thanks! I rebased the branch on master now. 

>Regarding the patches... they could simply mirror your commits.
I would do a single one with the content's of Akash's introductory patch and then additional ones with each of them adding some atomic feature like sound, tsadc and so on.

My patching skills are kinda non-existent, but I have given it a shot. I synced the boards  `current` branch to track `mainline's v5.5.y` and added Akash's patch to get a booting board. 
I still have to slowly attempt to migrate commits from [`ashthespy/linux-rockchip`](https://github.com/ashthespy/linux-rockchip/commits/5.5.y) over as a series of atomic patches as suggested..

igorpecovnik commented on config/bootenv/rockpis-dev.txt:
```diff
@@ -0,0 +1,3 @@
+verbosity=7
+extraargs=swiotlb=1024
+fdtfile='rockchip/rk3308-rock-pi-s.dtb'
```
board dt is not stored in u-boot environment? fdtfile has no default value?

ashthespy replied:
The u-boot env uses the legacy name of `rockpi-s-linux`. 
I initially created a patch for this - but then found this to work as well. 
```patch
diff --git a/configs/rockpi-s-rk3308_defconfig b/configs/rockpi-s-rk3308_defconfig
index c0cb439..f9d68f0 100644
--- a/configs/rockpi-s-rk3308_defconfig
+++ b/configs/rockpi-s-rk3308_defconfig
@@ -9,7 +9,7 @@ CONFIG_ROCKCHIP_VENDOR_PARTITION=y
 CONFIG_TARGET_EVB_RK3308=y
 CONFIG_TARGET_ROCKPIS_RK3308=y
 CONFIG_SPL_STACK_R_ADDR=0xc00000
-CONFIG_DEFAULT_DEVICE_TREE="rockpi-s-linux"
+CONFIG_DEFAULT_DEVICE_TREE="rk3308-rock-pi-s"
 CONFIG_DEBUG_UART=y
 CONFIG_ANDROID_BOOT_IMAGE=y
 CONFIG_FIT=y
```

piter75 replied:
@ashthespy I will fix it in the u-boot source to use the `rk3308-rock-pi-s` name for both legacy and dev.
We can then get rid of both `config/bootenv/rockpis-dev.txt` and `config/bootscripts/boot-rockpis-dev.cmd`.

I will give you a shout when it is done.

piter75 replied:
@ashthespy Done with those two commits: https://github.com/piter75/rockchip-u-boot/commit/cde3eb3c6ece39d3d3e69d57ecfa905a59fc6330 https://github.com/armbian/build/commit/992d0aa88cb2cacaf16660ce153dc51b7817b0c4
Now Rock Pi S' u-boot is expecting `/boot/dtb/rockchip/rk3308-rock-pi-s.dtb` file.

ashthespy replied:
All branches now use the same bootenv and bootscript, thanks!

ashthespy replied:
> @ashthespy Done with those two commits: [piter75/rockchip-u-boot@cde3eb3](https://github.com/piter75/rockchip-u-boot/commit/cde3eb3c6ece39d3d3e69d57ecfa905a59fc6330) [992d0aa](https://github.com/armbian/build/commit/992d0aa88cb2cacaf16660ce153dc51b7817b0c4)
> Now Rock Pi S' u-boot is expecting `/boot/dtb/rockchip/rk3308-rock-pi-s.dtb` file.

Just a heads up  [piter75/rockchip-u-boot@cde3eb3](https://github.com/piter75/rockchip-u-boot/commit/cde3eb3c6ece39d3d3e69d57ecfa905a59fc6330)  doesn't seem to work for me, u-boot still looks for `rockpi-s-linux.dtb` Or am I being daft?

piter75 replied:
> doesn't seem to work for me, u-boot still looks for `rockpi-s-linux.dtb` Or am I being daft?

It's not you. It is me ;p
I applied my knowledge of mainline u-boot to BSP one which behaves differently in this regard.
I fixed it in my BSP u-boot fork. It works well now.

ashthespy replied:
Prefect, thanks! :-) 

Side note: I tried playing with mainline u-boot, since there is support for rk3308 already, but wasn't able to get it to boot from my SD card, gotta dive back into that soon!

ashthespy commented:
@piter75 Did you manage to test this?

piter75 commented:
I am out of town right now - will test it over the weekend and give you a shout.

piter75 commented:
I finally found some time to test your mainline work for RockPi S.
I tested it both off your branch and on mainline with the patches you provided.

It is pretty usable already with console, storage and networking working.

igorpecovnik commented on config/boards/rockpi-s.conf:
```diff
@@ -2,5 +2,4 @@
 BOOTCONFIG="rockpi-s-rk3308_defconfig"
-KERNEL_TARGET="legacy"
-
+KERNEL_TARGET="legacy,current,dev"
```
I would propose to disable "current" here.

ashthespy replied:
I'm not extremely familiar with the build system - will disabling it here will still let you select the `current` branch from build script options? 
Basically this was my stop-gap solution while moving over from my branch to a patches

igorpecovnik replied:
> will disabling it here will still let you select the current branch from build script options?

No

> It is pretty usable already with console, storage and networking working.

The idea of DEV branch is to be in any, usually partially working, state. Current is "actual stable branch" and if this is so, I have not complains and we keep as is. Even its tied to non LTS kernel ATM.

piter75 commented:
I would be in favour of disabling current, tweaking the dev to use mainline kernel with the two patches provided by @ashthespy, merging it in and refining the support with following PRs.

What do you think @ashthespy?

ashthespy commented:
As you guys prefer - my patching game is weak and I had troubling getting them to stick, hence the separate repo with commits. I understand the hesitance to link to unknown repo, so I'll give porting the commits over a shot again in the coming weeks!

piter75 commented:
> my patching game is weak and I had troubling getting them to stick (...) I'll give porting the commits over a shot again in the coming weeks!

Try - it's always an opportunity to learn.
If you still encounter issues I will help you with the porting of your commits to patches :)

And... thanks for the contribution!

ashthespy commented:
Thanks for the offer, will trouble you when I run into issue! :-)
